### PR TITLE
add opTOO_MANY_SPONSORING tests

### PR DESCRIPTION
**Pull request by sisuresh**
# Checklist
- [ ] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [ ] Rebased on top of master (no merge commits)
- [ ] Ran `clang-format` v5.0.0 (via `make format` or the Visual Studio extension)
- [ ] Compiles
- [ ] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: src/transactions/test/CreateAccountTests.cpp [hunk 1]
```diff
@@ -198,4 +198,38 @@ TEST_CASE("create account", "[tx][createaccount]")
             }
         });
     }
+    SECTION("too many sponsoring")
+    {
+        auto a2 = root.create(
```
Comment (jonjove) [inline src/transactions/test/CreateAccountTests.cpp:203]: Not necessarily something you need to fix, just something I'm trying to understand. You specifically handle `CREATE_ACCOUNT` on SponsorshipTestUtils.cpp:330 but then you don't actually use that functionality here. Was this just a miss (perhaps you wrote this first) or did it not really fit into the broader functionality of that test (I've experienced that problem)?

File: src/transactions/test/CreateAccountTests.cpp [hunk 2]
```diff
@@ -198,4 +198,38 @@ TEST_CASE("create account", "[tx][createaccount]")
             }
         });
     }
+    SECTION("too many sponsoring")
+    {
```
Comment (jonjove) [inline src/transactions/test/CreateAccountTests.cpp:202]: Missing `for_versions_from(14, ...)`

File: src/transactions/test/CreateAccountTests.cpp (hunk 1 above)
Comment (sisuresh) [reply src/transactions/test/CreateAccountTests.cpp]: I originally wanted to use `tooManySponsoring` here, but the create account test was different enough that I decided not to use the helper. After looking at the the helper though, I realized that the create account use case didn't fit in well due to the way I setup the transactions. I used the sponsored account as the tx source account, which doesn't work for create account. Instead I could use the sponsoring account as the tx source account. 

I made this change, along with just using root at the sponsoring account to make things simpler. This allowed me to use `tooManySponsoring` for the create account case.

File: src/transactions/test/SponsorshipTestUtils.cpp [hunk 3]
```diff
@@ -291,4 +291,82 @@ createModifyAndRemoveSponsoredEntry(Application& app, TestAccount& sponsoredAcc,
...
+    tooManySponsoring(app, sponsoredAcc, sponsoredAcc, successfulOp, failOp);
+}
+void
+tooManySponsoring(Application& app, TestAccount& successfulOpAcc,
+                  TestAccount& failOpAcc, Operation const& successfulOp,
+                  Operation const& failOp)
+{
+    auto root = TestAccount::createRoot(app);
+    SECTION("too many sponsoring")
+    {
```
[the remark this answers is not in the document]
Comment (sisuresh) [reply src/transactions/test/SponsorshipTestUtils.cpp:309]: fixed

File: src/transactions/test/CreateAccountTests.cpp (hunk 2 above)
Comment (sisuresh) [reply src/transactions/test/CreateAccountTests.cpp:203]: This test uses `tooManySponsoring` now, which uses `for_versions_from(14, ...)`.

**Comment by MonsieurNicolas** [comment]
r+ 32c6e72b26cbd1f9b2b603b7b7e158386f3f32e5

**Pull request merged**